graviti/dataframe/dataframe.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+
+"""The implementation of the Graviti DataFrame."""
+
+
+from typing import Any, Dict, Iterable, Optional, Tuple, Union
+
+from graviti.dataframe.indexing import ILocIndexer
+from graviti.dataframe.series import Series
+
+
+class DataFrame:
+    """Two-dimensional, size-mutable, potentially heterogeneous tabular data.
+
+    Arguments:
+        data: The data that needs to be stored in DataFrame. Could be ndarray, Iterable or dict.
+            If data is a dict, column order follows insertion-order.
+        schema: The schema of the DataFrame. If None, will be inferred from `data`.
+        name: Name of the DataFrame.
+        copy: Copy data from inputs.
+            For dict data, the default of None behaves like ``copy=True``.  For DataFrame
+            or 2d ndarray input, the default of None behaves like ``copy=False``.
+        columns: Column labels to use for resulting frame when data does not have them,
+            defaulting to RangeIndex(0, 1, 2, ..., n). If data contains column labels,
+            will perform column selection instead.
+        client: The client for getting a remote data.
+
+    Examples:
+        Constructing DataFrame from a dictionary.
+
+        >>> data = { "col1": [1, 2], "col2": [3, 4] }
+        >>> df = pd.DataFrame(data=d)
+        >>> df
+           col1  col2
+        0     1     3
+        1     2     4
+
+        Constructing DataFrame from numpy ndarray:
+
+        >>> df2 = DataFrame(np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]]),
+        ...                    columns=["a", "b", "c"])
+        >>> df2
+           a  b  c
+        0  1  2  3
+        1  4  5  6
+        2  7  8  9
+
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        data: Union[Iterable[Any], Dict[Any, Any], "DataFrame", None] = None,
+        dtype: Any = None,
+        name: Optional[str] = None,
```

Comment:
Do we need to add a name here? Does this field do anything other than upload data? Whether it will add unnecessary trouble to future concat, merge methods.